Sources/Datadog/RUM/RUMMonitor/Scopes/RUMViewScope.swift

```diff
@@ -150,6 +150,9 @@ internal class RUMViewScope: RUMScope, RUMContextProvider {
         case let command as RUMAddUserActionCommand where isActiveView:
             if userActionScope == nil {
                 addDiscreteUserAction(on: command)
+            } else if command.actionType == .custom {
+                // still let it go, just instantly without any dependencies
+                sendDiscreteCustomUserAction(on: command)
             }
```

Comment:
I see this is also how we do it in `dd-sdk-android` 👌, but I'm wondering if this shouldn't be:
```swift
if command.actionType == .custom {
   sendDiscreteCustomUserAction(on: command)
} else if userActionScope == nil {
   addDiscreteUserAction(on: command)
}
```
?

This will uniformly treat all `.custom` actions as instant and send them right away. With the current implementation some `.custom` actions are managed on `userActionScope` (so have some duration) and some other do not (so have `0ns` duration). WDYT?

Reply:
On Android things are a bit different, there is no dedicated `AddCustomUserAction` command - instead there is `StartAction` with `waitForStop: Boolean` flag. In this case action will be sent only if in 100ms there was no activity. But I guess it is possible to have resource call in this 100ms, which makes an action added via `addAction` to have a dependency, so we cannot send it instantly.

So I'm not sure if we want to change this logic right now, instead current change is a bit safer and has less impact and just kind of `Plan B` if main branch didn't work.

I would say from the user point of view they would like to see resources attached to their custom actions, so it is better to keep it like that in my opinion.